vertx/src/main/java/com/microsoft/gctoolkit/vertx/jvm/AbstractJavaVirtualMachine.java

```diff
@@ -32,95 +29,112 @@
 
     private static final Logger LOGGER = Logger.getLogger(AbstractJavaVirtualMachine.class.getName());
 
-    private static final double LOG_FRAGMENT_THRESHOLD = 18000;
+    private static final double LOG_FRAGMENT_THRESHOLD = 60.0d;
 
-    private Diarizer jvmConfigurationFromParser;
-    private JvmConfiguration jvmConfigurationForCoreApi;
+    private Diary diary;
     private DateTimeStamp timeOfLastEvent;
     private final Map<Class<? extends Aggregation>, Aggregation> aggregatedData = new ConcurrentHashMap<>();
 
     @Override
     public boolean isG1GC() {
-        return jvmConfigurationFromParser.getDiary().isG1GC();
+        return diary.isG1GC();
     }
 
     @Override
     public boolean isZGC() {
-        return jvmConfigurationFromParser.getDiary().isZGC();
+        return diary.isZGC();
     }
 
     @Override
     public boolean isShenandoah() {
-        return jvmConfigurationFromParser.getDiary().isShenandoah();
+        return diary.isShenandoah();
     }
 
     @Override
     public boolean isParallel() {
-        return jvmConfigurationFromParser.getDiary().isParNew();
+        return diary.isParNew();
     }
 
     @Override
     public boolean isSerial() {
-        return jvmConfigurationFromParser.getDiary().isSerialFull();
+        return diary.isSerialFull();
     }
 
     @Override
     public boolean isCMS() {
-        return jvmConfigurationFromParser.getDiary().isCMS();
+        return diary.isCMS();
     }
 
     @Override
     public String getCommandLine() {
-        return jvmConfigurationFromParser.getCommandLine();
+        return ""; //todo: extract from diary... jvmConfigurationFromParser.getCommandLine();
     }
 
+    /**
+     * of the first event is significantly away from zero in relation to the time intervals between the
+     * of the next N events, where N maybe 1.
+     *
+     * try to estimate the time at which the JVM started. For log fragments, this will be the time
+     * of the first event in the log. Otherwise it will be 0.000 seconds.
+     * @return DateTimeStamp
+     */
+    @Override
+    public DateTimeStamp getEstimatedJVMStartTime() {
+        DateTimeStamp startTime = getTimeOfFirstEvent();
+        // Initial entries in GC log happen within seconds. Lets allow for 60 before considering the log
```

Comment:
60 is a magic comment, may e just ref the constant

Reply:
60 is indeed a magic number. This should be replaced with a heuristic. I'll add a comment